README.md

```diff
@@ -1,167 +1,198 @@
+This is not: rggr    (it must be an odd length)
+
+## Q1.1  (5 points)
+
+Write a Chomsky type 2 grammar that describes any valid path containing at
+least one paver. Use `S` as the start state, and `r`, `g`, `b` as terminals that
+represent the tiles.
+
+## A1.1
+
+S -> r| g | b
+S -> r S r
+S-> g S g
+S -> b S b
+
+r -> "r"
+g -> "g"
+b -> "b" 
+
+## Q1.2  (1 point for the O() answer, 2 for the sentence)
+
+Using big-O notation, what is the likely memory requirement for a parser that
+can validate a particular path configuration, where `n` is the number of tiles?
+In one sentence, explain why.
+
+## A1.2
+
+It would be O(n) because the parser has to check every input to determine if it is valid.
+
+
+# Q2
+
+A simple sentence structure might be
+
+    "the" (zero or more adjectives) noun verb (optional adverb)
+
+Adjectives are "lazy" or "smelly"
+
+Nouns are "dog" or "cat"
+
+Verbs are "ate" and "ran"
+
+Adverbs are "slowly" and "noisily"
+
+The following are examples of valid sentences:
+
+* The smelly dog ran.
+* The smelly dog ran slowly.
+* The cat ate noisily.
+
+## Q2.1 (7 points)
+
+
+
+## A2.1
+
+<sentence> ::= "the" <optional_adj> <noun> <verb> <adverb>
+
+<adj> ::= "lazy" | "smelly" 
+<optional_adj> ::= <adJ> | <adj> <optional_adj> | ""
+
+<noun> ::= "dog" | "cat"
+
+<verb> ::= "ate" | "ran"
+
+<adverb> ::= "slowly" | "noisily" | ""
+
+
+## Q2.2 (5 points)
+
+Write this grammar using EBNF with common extensions
+
+## A2.2
+
+<sentence> ::= "the" {<adj>} <noun> <verb> [<adverb>]
+<adjective> ::= "lazy" | "smelly"
+<noun> ::= "dog" | "cat"
+<verb> ::= "ate" | "ran"
+<adverb> ::= "slowly" | "noisily"
+
+
+## Q2.3 (6 points)
+
+  Draw a diagram for an FSM which recognizes these sentences. Use EOI as the
+  event that occurs at the end-of-input. Start with a state named S0. Receiving
+  the word "the" in that state will transition to state S1. Name the final state
+  END.
+
+  You can hand draw it and snap a picture, or use a drawing tool. Either way,
+  upload the image, and put the file name in the answer below.
+
+  (Hint: my answer has seven states including the start and end states)
+
+
+## A2.3
+
+The file is in github. It is called State Machine.jpg
+
+
+## Q2.4 (6 points)
+
+Convert this diagram into a table of the form:
+
+Current state | Next word | Next state
+--------------|-----------|-----------
+    S0        |    the    |     S1
+
+(hint: my version has 13 entries. Yours _might_ be different)
+
+## A2.4
+
+Current state | Next  word | Next state
+--------------|-----------|-----------
+    S0        |    the    |     S1
+    S1        |   lazy    |     S2
+    S1        |   smelly  |     S2
+    S1        |   dog     |     S3
+    S1        |   cat     |     S3
+    S2        |   lazy    |     S1
+    S2        |   smelly  |     S1
+    S2        |   ""      |     S1
+    S3        |   ate     |     S4
+    S3        |   ran     |     S4
+    S4        |   slowly  |     S5
+    S4        |   noisily |     S5

```

Comment:
# -3

This table is not a translation of your diagram. In the diagram. an adjective in S1 keeps you in S1. There's also the problem in the table where you introduce the rule on line 155 to get you back to S1. But you can't match nothing like that: how would the state machine know when to apply the rule.

I was looking for something like:

Current State | Next Word | Next State
------------- | --------- | ----------
S0            | "the"     | S1
S1            | "lazy"    | S1
S1            | "smelly"  | S1
S1            | "dog"     | S2
S1            | "cat"     | S2
S2            | "ate      | S3
S2            | "ran"     | S3
S3            | "slowly"  | S4
S3            | "noisily" | S4
S3            | EOI       | END
S4            | EOI       | END